OQS-OpenSSL_1_1_1-stable TEST70 test_clienthello FAIL.

The error message:

```
root@cbb4e2453d4c:/tmp/oqs_openssl# make test TESTS=test_clienthello  V=1
make depend && make _tests
make[1]: Entering directory '/tmp/oqs_openssl'
make[1]: Leaving directory '/tmp/oqs_openssl'
make[1]: Entering directory '/tmp/oqs_openssl'
( cd test; \
  mkdir -p test-runs; \
  SRCTOP=../. \
  BLDTOP=../. \
  RESULT_D=test-runs \
  PERL="/usr/bin/perl" \
  EXE_EXT= \
  OPENSSL_ENGINES=`cd .././engines 2>/dev/null && pwd` \
  OPENSSL_DEBUG_MEMORY=on \
    /usr/bin/perl .././test/run_tests.pl test_clienthello )
../test/recipes/70-test_clienthello.t .. 
1..1
    # Subtest: /tmp/oqs_openssl/test/clienthellotest
    1..1
        # Subtest: test_client_hello
        1..2
        ok 1 - iteration 1
        not ok 2 - iteration 2
    not ok 1 - test_client_hello
/tmp/oqs_openssl/util/shlib_wrap.sh /tmp/oqs_openssl/test/clienthellotest /tmp/oqs_openssl/test/session.pem => 1
not ok 1 - running clienthellotest
#   Failed test 'running clienthellotest'
#   at ../test/recipes/70-test_clienthello.t line 20.
# Looks like you failed 1 test of 1.
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 

Test Summary Report
-------------------
../test/recipes/70-test_clienthello.t (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=1, Tests=1,  0 wallclock secs ( 0.01 usr  0.01 sys +  0.06 cusr  0.01 csys =  0.09 CPU)
Result: FAIL
make[1]: *** [Makefile:208: _tests] Error 1
make[1]: Leaving directory '/tmp/oqs_openssl'
make: *** [Makefile:205: tests] Error 2

```

A Dockerfile to reproduce this:

 ```
FROM ubuntu:19.10

#Dependencies for OQS_OpenSSL
#We note that 'libssl-dev' is also a requirement, but
#we build that from source because we need a specific branch
RUN apt update && apt install -y \
	cmake                    \
	gcc                      \
	libtool                  \
	make                     \
	ninja-build              \
	unzip                    \
	xsltproc                 \
	git                      \
        linux-headers-$(uname -r)

#We clone all the repos along with the branches that we need
RUN cd /tmp                                                                                                         \
	&& git clone --branch OQS-OpenSSL_1_1_1-stable https://github.com/open-quantum-safe/openssl.git oqs_openssl \
	&& git clone --branch master                   https://github.com/open-quantum-safe/liboqs.git  oqs         \
	&& git clone --branch OpenSSL_1_1_1d	       https://github.com/openssl/openssl.git           openssl

#Store the install instances of OpenSSL before installing anything
RUN cd /tmp && find / -name libcrypto* > start_libcrypto.log
RUN cd /tmp && find / -name openssl    > start_openssl.log

#Build and install standard OpenSSL
RUN cd /tmp/openssl    \
	&& ./config    \
	&& make        \
	&& make test   \
	&& make install

#Store the install instances of OpenSSL after installing Standard OpenSSL
RUN cd /tmp && find / -name libcrypto* > after_std_libcrypto.log
RUN cd /tmp && find / -name openssl    > after_std_openssl.log

#Build and install OQS
RUN cd /tmp/oqs                                                          \
	&& mkdir build                                                   \
	&& cd build                                                      \
	&& cmake -GNinja -DCMAKE_INSTALL_PREFIX=../../oqs_openssl/oqs .. \
	&& ninja                                                         \
	&& ninja install

#Build and install OQS_OpenSSL
#RUN cd /tmp/oqs_openssl                             \
#	&& ./Configure no-shared linux-x86_64 -lm   \
#	&& make                                     \
#	&& make test                                \
#	&& make install


#Entrypoint
CMD [ "bash" ]
```

With this Dockerfile, run:
```
> docker build -t oqs_test .
> docker run -ti oqs_test bash
root@hash> cd /tmp/oqs_openssl/ && ./Configure no-shared linux-x86_64 -lm
root@hash> make 
root@hash> make test TESTS=test_clienthello  V=1
```

This error is consistent.  Helps. Thanks.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -26,7 +26,7 @@ localCheckout: &localCheckout
         command: .circleci/git_no_checkin_in_last_day.sh || (cd oqs_test && scripts/build_openssl.sh)
     - run:
         name: Run unit tests
-        command: mkdir -p oqs_test/test-results && .circleci/git_no_checkin_in_last_day.sh || (cd oqs_test && LD_LIBRARY_PATH="$(dirname $PWD);$(dirname $PWD)/oqs/lib" python3 -m nose --rednose --verbose --with-xunit --xunit-file=test-results/nosetests.xml)
+        command: mkdir -p oqs_test/test-results && .circleci/git_no_checkin_in_last_day.sh || (LD_LIBRARY_PATH="$(pwd):$(pwd)/oqs/lib" make test && cd oqs_test && LD_LIBRARY_PATH="$(dirname $PWD);$(dirname $PWD)/oqs/lib" python3 -m nose --rednose --verbose --with-xunit --xunit-file=test-results/nosetests.xml)
     - store_test_results: # Note that this command will fail when running CircleCI locally, that is expected behaviour
         path: oqs_test/test-results
 
@@ -90,6 +90,15 @@ jobs:
       LIBOQS: master
       OPENSSL: 111
       LIBTYPE: shared
+  ubuntu-bionic-x86_64-shared-noconn:
+    <<: *openssljob
+    environment:
+      IMAGE: openquantumsafe/ci-ubuntu-bionic-x86_64:latest
+      ARCH: x64
+      LIBOQS: master
+      OPENSSL: 111
+      LIBTYPE: shared
+      SKIP_TESTS: connection
   macOS-shared-noconn:
     <<: *macopenssljob
     environment:
@@ -110,6 +119,7 @@ workflows:
   build:
     jobs:
       - macOS-shared-noconn    
+      - ubuntu-bionic-x86_64-shared-noconn
       - ubuntu-bionic-x86_64-noconn
   nightly:
     triggers:

--- a/test/clienthellotest.c
+++ b/test/clienthellotest.c
@@ -27,7 +27,7 @@
 
 #define CLIENT_VERSION_LEN      2
 
-#define TOTAL_NUM_TESTS                        2// 4 see OQS note above
+#define TOTAL_NUM_TESTS                        1// 4 see OQS note above
 
 /*
  * Test that explicitly setting ticket data results in it appearing in the
@@ -35,7 +35,7 @@
  */
 #define TEST_SET_SESSION_TICK_DATA_VER_NEG      0
 /* Enable padding and make sure ClientHello is long enough to require it */
-#define TEST_ADD_PADDING                        1
+#define TEST_ADD_PADDING                       -3  // OQS creates ClientHello longer than this test ever envisaged
 /* Enable padding and make sure ClientHello is short enough to not need it */
 #define TEST_PADDING_NOT_NEEDED                 -1 // see OQS note above
 /*

